File: string/split-a-string-in-balanced-strings/split-a-string-in-balanced-strings.ts
function balancedStringSplit(s: string): number {
    let res = 0;
    let count = 0;

    for (let c of s) {
        count += (c === 'L' ? 1 : -1)
        if (count === 0) {
            res++;
        }
    }

    return res;
};

File: string/split-a-string-in-balanced-strings/split-a-string-in-balanced-strings.go
package splitastringinbalancedstrings

func balancedStringSplit(s string) int {
	res := 0

	count := 0
	for _, c := range s {
		if c == 'L' {
			count += 1
		} else {
			count -= 1
		}

		if count == 0 {
			res++
		}
	}

	return res
}
